src/GraphQL.Harness/Startup.cs

```diff
@@ -49,7 +49,18 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<EpisodeEnum>();
 
             // add schema
-            services.AddSingleton<ISchema, StarWarsSchema>();
+            services.AddSingleton<ISchema, StarWarsSchema>(services =>
+            {
+                var settings = services.GetRequiredService<IOptions<GraphQLSettings>>();
+                var schema = new StarWarsSchema(services);
+                if (settings.Value.EnableMetrics)
+                {
+                    schema.FieldMiddleware
+                        .Use(services.GetRequiredService<CountFieldMiddleware>())
+                        .Use(services.GetRequiredService<InstrumentFieldsMiddleware>());
+                }
+                return schema;
+            });
```

Comment:
If the schema constructor accepts `IEnumerable<IFieldMiddleware>`, then this example can be simplified.

Reply:
```
 // add Field Middlewares
            services.AddSingleton<CountFieldMiddleware>();
            services.AddSingleton<InstrumentFieldsMiddleware>();
```
->
```
 // add Field Middlewares
            services.TryAddEnumerable(ServiceDescriptor.Singleton<IFieldMiddleware, CountFieldsMiddleware>());
            services.TryAddEnumerable(ServiceDescriptor.Singleton<IFieldMiddleware, InstrumentFieldsMiddleware>());
```

Reply:
Well, you still need to pull `IOptions` from somewhere (unless you replace them with a primitive flag), but you can't do this in the `Configure` method itself, so I don't think thechanged example will be easier.

Reply:
I don't think we should have the `Schema` class's constructor accept `IEnumerable<IFieldMiddleware>`.  If we add that, what about `INameConverter`, `ISchemaFilter`, and `ISchemaComparer`?  Not to mention `Query`, `Mutation`, and `Subscription`.  Better just let the user inherit from `Schema` and add whatever they want to their constructor, as we can demonstrate in the documentation.

Reply:
I meant to add this ctor to the example, not to the main project.

Reply:
I see what you mean.  If we move it into the ctor then we need `GraphQLSettings` from the `Harness` project moved into the `StarWars` project and add a reference to the MS options package.  So I'm leaving it for now... but whatever you want is fine...

Reply:
Let's keep it as is. Perhaps later we publish more easier axample.